### PR TITLE
TGUI list input conversions

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
Independent of #63313 
- This converts several in game list inputs to tgui input list
- This sets tgui_input_list to respond to tgui_input being disabled in prefs

## Why It's Good For The Game
Better looking list inputs, much cleaner refactor
Tgui list input did not previously follow !tgui_inputs in prefs
Smaller batch of conversions to be safe

## Changelog

:cl:
fix: Turning off tgui inputs in game preferences now disables tgui input lists.
refactor: Converted several list inputs to tgui, including: Many AI functions, wayfinders, most cult inputs, observer functions.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm [hunk 1]
```diff
@@ -340,7 +340,7 @@
 
 	drawing = TRUE
 
-	var/type = pick_list[input(user,"Choose the rune","Rune") as null|anything in pick_list ]
+	var/type = pick_list[tgui_input_list(user, "Choose the rune", "Rune", pick_list) ]
```
Comment (jlsnow301) [inline code/modules/antagonists/eldritch_cult/eldritch_items.dm:343]: this looks horrendous

**Comment by jlsnow301** [comment]
Not really a mechanic change

**Comment by tralezab** [comment]
eh you know what you may have gotten verified but it isn't in the spirit of the verified label

File: code/modules/mob/dead/observer/observer.dm [hunk 2]
```diff
@@ -439,7 +439,7 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		var/area/A = V
 		if(!(A.area_flags & HIDDEN_AREA))
 			filtered += A
-	var/area/thearea = input("Area to jump to", "BOOYEA") as null|anything in filtered
+	var/area/thearea = tgui_input_list(usr, "Area to jump to", "Area Hop", filtered)
```
Comment (Sealed101) [inline code/modules/mob/dead/observer/observer.dm:442]: BOOYEA my beloved

Comment (jlsnow301) [reply code/modules/mob/dead/observer/observer.dm]: resouled

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/antagonists/eldritch_cult/eldritch_items.dm:343]: Yeah

**Pull request merged**